systems/subscriber/sim-pool/pkg/server/sim_test.go

```diff
@@ -25,187 +26,418 @@ import (
+		res, err := simService.GetStats(context.Background(), reqMock)
+		assert.Error(t, err)
+		assert.Nil(t, res)
+		mockRepo.AssertExpectations(t)
+	})
 }
 
-func TestGetStats_Error(t *testing.T) {
-	mockRepo := &mocks.SimRepo{}
-	msgbusClient := &mbmocks.MsgBusServiceClient{}
-	simService := NewSimPoolServer(OrgName, mockRepo, msgbusClient)
-	reqMock := &pb.GetStatsRequest{
-		SimType: "ukama_data",
-	}
-	mockRepo.On("GetSimsByType", mock.Anything).Return(nil, grpc.SqlErrorToGrpc(errors.New("SimPool record not found!"), "sim-pool"))
-	res, err := simService.GetStats(context.Background(), reqMock)
-	assert.Error(t, err)
-	assert.Nil(t, res)
-	mockRepo.AssertExpectations(t)
+func TestDelete(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		mockRepo := &mocks.SimRepo{}
+		msgbusClient := &mbmocks.MsgBusServiceClient{}
+		simService := NewSimPoolServer(OrgName, mockRepo, msgbusClient)
+		reqMock := &pb.DeleteRequest{
+			Id: []uint64{1},
+		}
+		mockRepo.On("Delete", mock.Anything).Return(nil)
+		msgbusClient.On("PublishRequest", mock.AnythingOfType("string"), mock.AnythingOfType("*events.SimRemoved")).Return(nil).Once()
+		res, err := simService.Delete(context.Background(), reqMock)
+		assert.NoError(t, err)
+		assert.Equal(t, reqMock.Id[0], res.Id[0])
+
+		mockRepo.AssertExpectations(t)
+		msgbusClient.AssertExpectations(t)
+	})
+
+	t.Run("Error", func(t *testing.T) {
+		mockRepo := &mocks.SimRepo{}
+		msgbusClient := &mbmocks.MsgBusServiceClient{}
+		simService := NewSimPoolServer(OrgName, mockRepo, msgbusClient)
+		reqMock := &pb.DeleteRequest{
+			Id: []uint64{1},
+		}
+		mockRepo.On("Delete", mock.Anything).Return(grpc.SqlErrorToGrpc(errors.New("Error while deleting record!"), "sim-pool"))
+		res, err := simService.Delete(context.Background(), reqMock)
+		assert.Error(t, err)
+		assert.Nil(t, res)
+		mockRepo.AssertExpectations(t)
+	})
 }
 
-func TestDelete_Success(t *testing.T) {
-	mockRepo := &mocks.SimRepo{}
-	msgbusClient := &mbmocks.MsgBusServiceClient{}
-	simService := NewSimPoolServer(OrgName, mockRepo, msgbusClient)
-	reqMock := &pb.DeleteRequest{
-		Id: []uint64{1},
-	}
-	mockRepo.On("Delete", mock.Anything).Return(nil)
-	msgbusClient.On("PublishRequest", mock.AnythingOfType("string"), mock.AnythingOfType("*events.SimRemoved")).Return(nil).Once()
-	res, err := simService.Delete(context.Background(), reqMock)
-	assert.NoError(t, err)
-	assert.Equal(t, reqMock.Id[0], res.Id[0])
-
-	mockRepo.AssertExpectations(t)
-	msgbusClient.AssertExpectations(t)
+func TestAdd(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		mockRepo := &mocks.SimRepo{}
+		msgbusClient := &mbmocks.MsgBusServiceClient{}
+		simService := NewSimPoolServer(OrgName, mockRepo, msgbusClient)
+		reqMock := &pb.AddRequest{
+			Sim: []*pb.AddSim{
+				{
+					Iccid:          "1234567890123456789",
+					Msisdn:         "2345678901",
+					SimType:        "ukama_data",
```

Comment:
Use `ukama.SimTypeUkamaData.String()` instead of hard coding `ukama_data` value.